proof.go

```diff
@@ -206,10 +206,19 @@ func (proof Proof) verifyLeafHashes(nth *Hasher, verifyCompleteness bool, nID na
 	return bytes.Equal(tree.Root(), root)
 }
 
-func (proof Proof) VerifyInclusion(h hash.Hash, nid namespace.ID, data []byte, root []byte) bool {
+// VerifyInclusion checks that the inclusion proof is valid by using leaf data
+// and the provided proof to regenerate and compare the root. Note that the leaf
+// data should not contain the prefixed namespace, unlike the tree.Push method,
+// which takes prefixed data. All leaves implicitly have the same namespace ID: `nid`.
+func (proof Proof) VerifyInclusion(h hash.Hash, nid namespace.ID, leaves [][]byte, root []byte) bool {
```

Comment:
An additional convenience method `VerifyNamespaceRange` or sth alike does not make sense?

Reply:
I can see that working as well, tbh I just didn't think about that and was just following the underlying library's API.

I want to merge to wrap this up, but I also don't want to lose this comment, so #59